src/main/java/com/mercuryCyclists/Sales/service/InStoreSaleService.java

```diff
@@ -0,0 +1,182 @@
+        // Validate sale
+        if (!inStoreSale.validate()) {
+            throw new IllegalStateException("Invalid In Store Sale");
+        }
+
+        // Do after first validation in case quantity is null
+        if (inStoreSale.getQuantity() <= 0) {
+            throw new IllegalStateException("In Store Sale has 0 or a negative quantity");
+        }
+
+        // Check if store exists
+        Optional<Store> store = storeRepository.findById(storeId);
+        if (!store.isPresent()) {
+            throw new IllegalStateException(String.format("Store with Id %s does not exist", storeId));
+        }
+
+        // Check product exists
+        JsonObject product = saleService.getSaleProduct(inStoreSale);
+        if(product.get("id") == null){
+            throw new IllegalArgumentException(String.format("Invalid product, %s", product));
+        }
+
+        // If there is enough of the product is stock
+        Long productQuantity = product.get("quantity").getAsLong();
+        Long saleQuantity = inStoreSale.getQuantity();
+        if (productQuantity >= saleQuantity) {
+            // Update and save product
+            product.addProperty("quantity", (productQuantity - saleQuantity));
+            saleService.updateProduct(product);
+
+            // Save and return sale
+            inStoreSale.setStore(store.get());
+            inStoreSaleRepository.save(inStoreSale);
+            return new ResponseEntity<>(inStoreSale, HttpStatus.CREATED);
+        } else {
+            // Get product parts
+            JsonArray productParts = saleService.getSaleProductParts(inStoreSale);
+            ArrayList<JsonObject> partJsonObjs = new ArrayList<JsonObject>();
+
+            // For each part in the product
+            for (JsonElement part : productParts) {
+                // Convert json element to json object
+                JsonObject partJsonObj = part.getAsJsonObject();
+
+                Long partQuantity = partJsonObj.get("quantity").getAsLong();
+                if (partQuantity >= saleQuantity) {
+                    // Update part quantity
+                    partJsonObj.addProperty("quantity", (partQuantity - saleQuantity));
+
+                    // Save to ArrayList of JsonObjects
+                    partJsonObjs.add(partJsonObj);
+                } else {
+                    // return 303 Error
+                    return new ResponseEntity<>(inStoreSale, HttpStatus.SEE_OTHER);
+                }
+            }
+
+            // For each json object part in json object part array list
+            // Save the json object part
+            for (JsonObject part : partJsonObjs) {
+                saleService.updateProductPart(part);
+            }
+
+            // Save and return sale
+            inStoreSale.setStore(store.get());
+            inStoreSaleRepository.save(inStoreSale);
+            return new ResponseEntity<>(inStoreSale, HttpStatus.CREATED);
+        }
```

Comment:
Nice looks good!
I would recommend splitting into 2 different methods for product and part, then chucking it in the sale service.